tests/foreman/ui/test_repository.py

```diff
@@ -234,27 +228,28 @@ def test_positive_update_URL(self, repo_name):
         """
         product = entities.Product(organization=self.organization).create()
         with Session(self.browser) as session:
-            make_repository(
-                session,
-                org=self.organization.name,
-                name=repo_name,
-                product=product.name,
-                url=FAKE_1_YUM_REPO,
-            )
-            self.assertIsNotNone(self.repository.search(repo_name))
-            self.assertTrue(self.repository.validate_field(
-                repo_name, 'url', FAKE_1_YUM_REPO))
-            self.navigator.go_to_products()
-            self.products.search(product.name).click()
-            self.repository.update(repo_name, new_url=FAKE_2_YUM_REPO)
-            self.navigator.go_to_products()
-            self.products.search(product.name).click()
-            self.assertTrue(self.repository.validate_field(
-                repo_name, 'url', FAKE_2_YUM_REPO))
+            for repo_name in generate_strings_list():
+                with self.subTest(repo_name):
+                    make_repository(
+                        session,
+                        org=self.organization.name,
+                        name=repo_name,
+                        product=product.name,
+                        url=FAKE_1_YUM_REPO,
+                    )
+                    self.assertIsNotNone(self.repository.search(repo_name))
+                    self.assertTrue(self.repository.validate_field(
+                        repo_name, 'url', FAKE_1_YUM_REPO))
+                    self.navigator.go_to_products()
+                    self.products.search(product.name).click()
+                    self.repository.update(repo_name, new_url=FAKE_2_YUM_REPO)
+                    self.navigator.go_to_products()
+                    self.products.search(product.name).click()
+                    self.assertTrue(self.repository.validate_field(
+                        repo_name, 'url', FAKE_2_YUM_REPO))
 
     @run_only_on('sat')
-    @data(*generate_strings_list())
-    def test_positive_update_GPG(self, repo_name):
+    def test_positive_update_GPG(self):
```

Comment:
I left test_positive_update_URL as datadriven and changed others to normal tests to avoid test redundancy.